File: 72.py
'''
@description: 百钱买百鸡的问题
@param {type} 
@return: 
'''
# 直接输出
while True:
    try:
        a = input()
        print("0 25 75")
        print("4 18 78")
        print("8 11 81")
        print("12 4 84")
    except:
        break

# 计算
while True:
    try:
        k=input()
        for a in range(21):
            for b in range(33):
                c=(100-5*a-3*b)*3
                if c>=0 and a+b+c==100:
                    print(str(a)+' '+str(b)+' '+str(c))
    except:
        break

File: 72.c

/**
 * @description: 百钱买白鸡的问题
 * @param {type} 
 * @return: 
 */
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>

int testnum;

int main(void)
{
    while(scanf("%d", &testnum) != EOF)
    {
        //计算
        for(int i = 0; i <= 20; i++)
        {
            for(int j = 0; j <= 33; j++)
            {
                if(((100-i-j)%3==0)&&((i*5+j*3+(100-i-j)/3)==100))
                {
                    printf("%d %d %d\n",i,j,(100-i-j));
                    break;
                }
            }
        }
        //直接输出
        // printf("0 25 75\n4 18 78\n8 11 81\n12 4 84\n");
    }

    return 0;
}
